messenger.rst

```diff
@@ -654,35 +736,181 @@ This middleware can be configured to define the entity manager to use:
+under the transport in ``messenger.yaml``:
+
+==================  =================================== =======
+     Option               Description                   Default
+==================  =================================== =======
+stream              The Redis stream name               messages
+group               The Redis consumer group name       symfony
+consumer            Consumer name used in Redis         consumer
+serializer          How to serialize the final payload  ``Redis::SERIALIZER_PHP``
+                    in Redis (the
+                    ``Redis::OPT_SERIALIZER`` option)
+==================  =================================== =======
+
+in-memory Transport
+~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.3
+
+    The ``in-memory`` transport was introduced in Symfony 4.3.
+
+The ``in-memory`` transport does not actually delivery messages. Instead, it
+holds them in memory during the request, which can be useful for testing.
+For example, if you have an ``async_priority_normal`` transport, you could
+override it in the ``test`` environment to use this transport:
+
+.. code-block:: yaml
+
+    # config/packages/test/messenger.yaml
+    framework:
+        messenger:
+            transports:
+                async_priority_normal: 'in-memory:///'
+
+Then, while testing, messages will *not* be delivered to the real transport.
+Even better, in a test, you can check that exactly one message was sent
+during a request::
+
+    // tests/DefaultControllerTest.php
+    namespace App\Tests;
+
+    use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+    use Symfony\Component\Messenger\Transport\InMemoryTransport;
+
+    class DefaultControllerTest extends WebTestCase
+    {
+        public function testSomething()
+        {
+            $client = static::createClient();
+            // ...
+
+            $this->assertSame(200, $client->getResponse()->getStatusCode());
+
+            /* @var InMemoryTransport $transport */
+            $transport = self::$container->get('messenger.transport.async_priority_normal');
+            $this->assertCount(1, $transport->get());
+        }
+    }
+
+Serializing Messages
+~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.3
+    The default serializer changed in 4.3 from the Symfony serializer to the
+    native PHP serializer. Existing applications should configure their transports
+    to use the Symfony serializer to avoid losing already-queued messages after
+    upgrading.
+
+When messages are sent to (and received from) a transport, they're serialized
+using PHP's native ``serialize()`` & ``unserialize()`` functions. You can change
+this globally (or for each transport) to a service that implements
+:class:`Symfony\\Component\\Messenger\\Transport\\Serialization\\SerializerInterface`:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/messenger.yaml
+        framework:
+            messenger:
+                serializer:
```

Comment:
We can also add this example in the recipe too ? WDYT

Reply:
I think most people should not need to worry about the serializer or changing its config. It could prove that I'm wrong about that over time, but if I'm right, I think it's better to not make this config "crowd" the more important config in the recipe